packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.tsx

```diff
@@ -0,0 +1,77 @@
+import { Language } from '@logto/phrases';
+import React, { useMemo, useState, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import Dropdown, { DropdownItem } from '@/components/Dropdown';
+import useSocial from '@/hooks/use-social';
+import { ConnectorData } from '@/types';
+
+import * as styles from './SocialSignInDropdown.module.scss';
+
+type Props = {
+  anchorRef?: React.RefObject<HTMLElement | SVGSVGElement>;
+  isOpen: boolean;
+  onClose: () => void;
+  connectors: ConnectorData[];
+};
+
+const SocialSignInDropdown = ({ isOpen, onClose, connectors, anchorRef }: Props) => {
+  const {
+    i18n: { language },
+  } = useTranslation();
+
+  const { invokeSocialSignIn } = useSocial();
+
+  const [contentStyle, setContentStyle] = useState<{ top?: number; left?: number }>();
+
+  const items = useMemo(
+    () =>
+      connectors.map(({ id, name, logo }) => {
+        const languageKey = Object.keys(name).find((key) => key === language) ?? 'en';
+        const localName = name[languageKey as Language];
+
+        return (
+          <DropdownItem
+            key={id}
+            onClick={() => {
+              void invokeSocialSignIn(id, onClose);
+            }}
+          >
+            <img src={logo} alt={id} className={styles.social_logo} />
+            <span>{localName}</span>
+          </DropdownItem>
+        );
+      }),
+    [connectors, language, invokeSocialSignIn, onClose]
+  );
+
+  const adjustPosition = useCallback(() => {
+    if (anchorRef?.current) {
+      const { left, top } = anchorRef.current.getBoundingClientRect();
+
+      setContentStyle({
+        left,
+        top: top - 8,
+      });
+    }
+  }, [anchorRef]);
+
+  return (
+    <Dropdown
+      role="dropdown"
+      isOpen={isOpen}
+      className={styles.socialDropDown}
+      style={{ content: contentStyle }}
```

Comment:
Just curious: Isn't `content only` available in `:before` or `:after` pseudo classes?

Reply:
This attribute is ReactModal.style, which accepts two params content and overlay.  not HTML style

Reply:
ok